app/app.go

```diff
@@ -0,0 +1,77 @@
+package app
+
+import (
+	"log"
+	"os"
+
+	"github.com/square/etre/auth"
+	"github.com/square/etre/cdc"
+	"github.com/square/etre/config"
+	"github.com/square/etre/entity"
+	"github.com/square/etre/metrics"
+)
+
+type Context struct {
+	ConfigFile string
+	Config     config.Config
+
+	EntityStore     entity.Store
+	EntityValidator entity.Validator
+	CDCStore        cdc.Store
+	FeedFactory     cdc.FeedFactory
+	MetricsStore    metrics.Store
+	MetricsFactory  metrics.Factory
+	Auth            auth.Manager
+
+	// 3rd-party extensions, all optional
+	Hooks   Hooks
+	Plugins Plugins
+}
+
+type Hooks struct {
+	LoadConfig func(Context) (config.Config, error)
+	RunAPI     func() error
+	StopAPI    func() error
```

Comment:
A comment explaining RunAPI and StopAPI would be helpful.

Reply:
Done.